### PR TITLE
doc: Bluetooth: mesh: add LCD documentation

### PR DESCRIPTION
**Pull request by akredalen**
Added basic description of Large Composition Data server and client models.

### CODE REVIEW

**Review by m-alperen-sener — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/lcd_cli.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+The Large Composition Data Client model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_CLI` option.
+
+The Large Composition Data Client model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and supports the functionality
+of reading pages of composition data that do not fit in a Config Composition Data Status message and reading the metadata of the model instances on a node that supports the :ref:`bluetooth_mesh_lcd_srv` model.
+
+The Large Composition Data Client model communicates with a Large Composition Data Server model using the device key of the node containing the target Large Composition Data Server model instance.
+
+If present, the Large Composition Data Client model must be instantiated on the primary element.
```
Comment (Andrewpini) [inline doc/connectivity/bluetooth/api/mesh/lcd_cli.rst:14]:
```suggestion
If present, the Large Composition Data Client model must only be instantiated on the primary element.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Metadata contains information about the models instantiated on the element. The :c:member:`bt_mesh_models_metadata_entry` struct can be used to define a list of metadata entries.
+Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and id of the metadata, and a pointer to the raw data.
+If the list is empty, the helper macro `BT_MESH_MODELS_METADATA_NONE` can be used instead.
+The `BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.
+
+The Large Composition Data Server does not have an API of its own and relies on a :ref:`bluetooth_mesh_lcd_cli` to control it.
+The model only accepts messages encrypted with the node's device key.
+
+If present, the Large Composition Data Server model must be instantiated on the primary element.
```
Comment (Andrewpini) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:20]:
```suggestion
If present, the Large Composition Data Server model must only be instantiated on the primary element.
```

**Review by Andrewpini — CHANGES_REQUESTED** [review]
Looks fine to me, got a couple of minor suggestions..

**Comment by akredalen** [comment]
@mia-ko could you have a look?

**Comment by mia-ko** [comment]
> @mia-ko could you have a look?

Yup, will do :)

File: doc/connectivity/bluetooth/api/mesh/lcd_cli.rst [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.. _bluetooth_mesh_lcd_cli:
+
+Large Composition Data Client
+#############################
+
+The Large Composition Data Client model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_CLI` option.
+
+The Large Composition Data Client model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and supports the functionality
+of reading pages of composition data that do not fit in a Config Composition Data Status message and reading the metadata of the model instances on a node that supports the :ref:`bluetooth_mesh_lcd_srv` model.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_cli.rst:10]:
```suggestion
of reading pages of Composition Data that do not fit in a Config Composition Data Status message and reading the metadata of the model instances on a node that supports the :ref:`bluetooth_mesh_lcd_srv` model.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.. _bluetooth_mesh_lcd_srv:
+
+Large Composition Data Server
+#############################
+
+The Large Composition Data Server model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` option.
+
+The Large Composition Data Server model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and is used to support
+the functionality of exposing pages of composition data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:10]:
```suggestion
the functionality of exposing pages of Composition Data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+Metadata contains information about the models instantiated on the element. The :c:member:`bt_mesh_models_metadata_entry` struct can be used to define a list of metadata entries.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:12]:
```suggestion
Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and id of the metadata, and a pointer to the raw data.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:13]:
```suggestion
Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_cli.rst [hunk 7]
```diff
@@ -0,0 +1,21 @@
+.. _bluetooth_mesh_lcd_cli:
+
+Large Composition Data Client
+#############################
+
+The Large Composition Data Client model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_CLI` option.
+
+The Large Composition Data Client model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and supports the functionality
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_cli.rst:9]: This line break looks a bit odd.

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+The Large Composition Data Server model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` option.
+
+The Large Composition Data Server model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and is used to support
+the functionality of exposing pages of Composition Data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
+
+Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
+Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
+If the list is empty, the helper macro `BT_MESH_MODELS_METADATA_NONE` can be used instead.
+The `BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:15]: I somehow managed to miss this yesterday... Shouldn't these macros be written with :c:macro:`BT_MESH_MODELS_METADATA_ENTRY`, etc?

Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:15]: These comments render strange sometimes. I am using ` in front and end of the name.

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+The Large Composition Data Server model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` option.
+
+The Large Composition Data Server model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and is used to support
+the functionality of exposing pages of Composition Data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
+
+Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
+Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
+If the list is empty, the helper macro `BT_MESH_MODELS_METADATA_NONE` can be used instead.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:14]:
```suggestion
If the list is empty, the helper macro :c:macro:`BT_MESH_MODELS_METADATA_NONE` can be used instead.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+#############################
+
+The Large Composition Data Server model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` option.
+
+The Large Composition Data Server model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and is used to support
+the functionality of exposing pages of Composition Data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
+
+Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
+Entries should be created using the `BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:13]:
```suggestion
Entries should be created using the :c:macro:`BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
```

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst (hunk 8 above)
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:15]:
```suggestion
The :c:macro:`BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/lcd_srv.rst [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+The Large Composition Data Server model is a foundation model defined by the Bluetooth
+mesh specification. The model is optional, and is enabled through the :kconfig:option:`CONFIG_BT_MESH_LARGE_COMP_DATA_SRV` option.
+
+The Large Composition Data Server model was introduced in the Bluetooth Mesh Protocol Specification version 1.1, and is used to support
+the functionality of exposing pages of Composition Data that do not fit in a Config Composition Data Status message and to expose metadata of the model instances.
+
+Metadata contains information about the models instantiated on the element. The structure member :c:member:`bt_mesh_models_metadata_entry` can be used to define a list of metadata entries.
+Entries should be created using the :c:macro:`BT_MESH_MODELS_METADATA_ENTRY` macro. Each entry contains the length and ID of the metadata, and a pointer to the raw data.
+If the list is empty, the helper macro :c:macro:`BT_MESH_MODELS_METADATA_NONE` can be used instead.
+The :c:macro:`BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present.
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/lcd_srv.rst:15]:
```suggestion
Models metadata
===============

The Large Composition Data Server model allows each model to have a list of model's specific metadata that can be read by the Large Composition Data Client model. The metadata list can be associated with the :c:struct:`bt_mesh_model` through the :c:member:`bt_mesh_model.metadata` field. The metadata list consists of one or more entries defined by the :c:struct:`bt_mesh_models_metadata_entry` structure. Each entry contains the length and ID of the metadata, and a pointer to the raw data. Entries can be created using the :c:macro:`BT_MESH_MODELS_METADATA_ENTRY` macro. The :c:macro:`BT_MESH_MODELS_METADATA_END` macro marks the end of the metadata list and must always be present. If the model has no metadata, the helper macro :c:macro:`BT_MESH_MODELS_METADATA_NONE` can be used instead.
```

And move it below line 20.

**Review by PavelVPV — APPROVED** [review]

**Review by mia-ko — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Pull request merged**